README.markdown

```diff
@@ -972,6 +974,40 @@ if let number1 = number1 {
 
 Guard statements are required to exit in some way. Generally, this should be simple one line statement such as `return`, `throw`, `break`, `continue`, and `fatalError()`. Large code blocks should be avoided. If cleanup code is required for multiple exit points, consider using a `defer` block to avoid cleanup code duplication.
 
+### Defensive Techniques
+
+Use `sweetAssertionFailure` to catch errors / issues as early as possible.
+**Preferred**:
+```swift
+guard let number1 = number1 else {
+    sweetAssertionFailure("Number should not be nil")
+}
+// do something with number
+```
+
+**Not Preferred**:
+```swift
+guard let number1 = number1 else {
+    return
+}
+// do something with number
+```
+
+Use `safe` for array indexing without crashes.
```

Comment:
`safe` is useful in cases of a fix index, not necessary 0, because here we have also the `.first` var.
In case I have a loop inside the array, I prefer guarding the min, max of the loop, before starting a loop.

Reply:
Maybe we should add this in another PR as this one is already accepted?